internal/training/grpo_engine/lora.c

```
@@ -0,0 +1,318 @@
+int lora_save(LoRAEngine *le, const char *path, int step, float loss) {
+    FILE *f = fopen(path, "wb");
+    if (!f) return -1;
+
+    /* Header: 64 bytes */
+    char header[64] = {0};
+    memcpy(header, DCLORA_MAGIC, 8);
+    *(uint32_t *)(header + 8) = (uint32_t)le->n_layers;
+    *(uint32_t *)(header + 12) = (uint32_t)le->rank;
+    *(uint32_t *)(header + 16) = MAX_TARGETS;
+    *(uint32_t *)(header + 20) = (uint32_t)step;
+    *(float *)(header + 24) = loss;
```

Cppcheck (2.21.0)

[portability] Casting between char * and float * which have an incompatible binary data representation.

(invalidPointerCast)